exist-core/src/main/java/org/exist/xquery/functions/fn/FunPath.java

```diff
@@ -0,0 +1,226 @@
+                    // if the root node is not a document node.
+                    pathValues.remove(0);
+                    result = new StringValue(String.format("Q{%s}root()", Namespaces.XPATH_FUNCTIONS_NS) + String.join("", pathValues));
+                } else {
+                    result = new StringValue(String.join("", pathValues));
+                }
+            } else {
+                // If the context item is not a node, type error [err:XPTY0004].
+                throw new XPathException(this, ErrorCodes.XPTY0004,  "Item is not a document or node; got '" + item + "'", sequence);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Gets the position of a specified node among its like-named siblings.
+     *
+     * @param   node  the node whose position to get
+     * @return  the position of the specified node, or zero if this method
+     *          failed to determine the position of the specified node
+     */
+    private static int getNodePosition(final Node node) {
+        int position = 1;
+        Node siblingNode = node.getPreviousSibling();
+        while (siblingNode != null) {
+            if (siblingNode != null && siblingNode.getNodeName().equals(node.getNodeName())) {
+                ++position;
+            }
+            siblingNode = siblingNode.getPreviousSibling();
+        }
+        return position;
+    }
+
+    /**
+     * Gets the path values of a specified node.
+     *
+     * @param   node    the node whose path values to get
+     * @param   values  the path values
+     */
+    private static void getPathValues(Node node, final List<String> values) {
+        // get the owner element, if the specified node is an attribute node
+        Node attributeNode;
+        if (node.getNodeType() == Type.ATTRIBUTE) {
+            attributeNode = node;
+            node = ((Attr) node).getOwnerElement();
+        } else {
+            attributeNode = null;
+        }
+
+        final StringBuilder value = new StringBuilder();
+        final StringBuilder attributeValue = new StringBuilder();
+        if (node.getNodeType() == Type.TEXT) {
+            // For a text node: text()[position] where position is an integer
+            // representing the position of the selected node among its text
+            // node siblings
+            final int nodePosition = getNodePosition(node);
+            if (nodePosition > 0) {
+                value.append(String.format("/text()[%d]", nodePosition));
+            }
+        } else if (node.getNodeType() == Type.COMMENT) {
+            // For a comment node: comment()[position] where position is an
+            // integer representing the position of the selected node among
+            // its comment node siblings.
+            final int nodePosition = getNodePosition(node);
+            if (nodePosition > 0) {
+                value.append(String.format("/comment()[%d]", nodePosition));
```

Comment:
is the 1st position nodePosition 0 or 1 ?
does the XQTS not require `/comment()`  (without indication) to be used iso `/comment()[1]` ?

Reply:
This is passing all XQTS tests for fn-path